### PR TITLE
Db merge obelisk 2019 01 20

### PR DESCRIPTION
**Pull request by danbornside**
some fixes that seem to make some things hang a bunch less

### CODE REVIEW

File: gargoyle-postgresql/src/Gargoyle/PostgreSQL.hs [hunk 1]
```diff
@@ -74,53 +74,66 @@ getLocalPostgresConnectionString dbDir = do
...
+                         -> IO ()
+shutdownPostgresWithMode mode binPath absoluteDbDir = do
+  (_, _, _, postgres) <- createProcess (proc binPath
+    [ "stop"
+    , "-D", absoluteDbDir
+    , "-w"
+    , "-m", mode
+    ]) { std_in = NoStream, std_out = NoStream, std_err = Inherit }
+  r <- waitForProcess postgres
+  case r of
```
Comment (3noch) [inline gargoyle-postgresql/src/Gargoyle/PostgreSQL.hs]: I think this error reporting should be in a `finally` clause.

Comment (3noch) [reply gargoyle-postgresql/src/Gargoyle/PostgreSQL.hs]: Same elsewhere

Comment (3noch) [reply gargoyle-postgresql/src/Gargoyle/PostgreSQL.hs]: Although, with `exitWith` that would also cause executable to exit in the case of an async exception to the thread. Is that what we want?

**Comment by ryantrinkle** [comment]
I believe this is obsoleted by the changes I merged in #25.

**Pull request closed without merging**